contracts/tonomy/include/tonomy/tonomy.hpp

```diff
@@ -268,6 +292,23 @@ namespace tonomysystem
       // Following line needed to correctly generate ABI. See https://github.com/EOSIO/eosio.cdt/issues/280#issuecomment-439666574
       typedef eosio::multi_index<"resconfig"_n, resource_config> resource_config_table_dump;
 
+      // Define the structure of a staking allocation
+      struct [[eosio::table]] staking_allocation
+      {
+         uint64_t id;
+         eosio::name account_name; // The account name of the staker.
+         eosio::asset tokens_staked; //The amount of tokens staked.
+         eosio::time_point stake_time; //The time when the staking started.
+         eosio::time_point unstake_time; //The time when the unstaking will occur.
+         bool unstake_requested; //A flag indicating whether the tokens are currently being unstaked.
```

Comment:
I would like us to add a `int staking_category` field here to make this easier to upgrade in the future. this should be set to `1` for now, but can be used later to change users to different staking systems.